### PR TITLE
Add Semi-SL Instance Segmentation Documentation

### PR DESCRIPTION
**Pull request by kprokofi**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: docs/source/guide/explanation/algorithms/segmentation/instance_segmentation.rst [hunk 1]
```diff
@@ -92,11 +92,52 @@ Recently, we have made updates to MaskRCNN-ConvNeXt, incorporating the `ConvNeXt
...
+This means that backward loss propagation is not utilized for updating the teacher model. Once training is complete, only the student model is used for making predictions in the instance segmentation task.
+
+We also use the warmup stage for the teacher model during the first epochs to avoid utilizing too misleading pseudo labeling. We use constant thresholding for pseudo bounding boxes and, unlike in Unbiased Teacher work, we utilize all unlabeled losses including regression one and mask loss.
+There are some key differences in the augmentation pipelines used for labeled and unlabeled data.
+Basic augmentations such as random flip, random rotate, and random crop are employed for the teacher model's input.
+On the other hand, stronger augmentations like color distortion, RGB to gray conversion, Gaussian blur and `Random Erasing <https://arxiv.org/abs/1708.04896>`_ are applied to the student model.
+This discrepancy helps improve generalization and prevents unnecessary overfitting on the pseudo-labels generated by the teacher model.
+As well as for the supervised pipeline we utilize EMA smoothing for the student model during warm-up and the rest of the training.
```
Comment (sovrasov) [inline docs/source/guide/explanation/algorithms/segmentation/instance_segmentation.rst]: In the same way as ...
... for the student model throughout the whole training process

File: docs/source/guide/explanation/algorithms/segmentation/instance_segmentation.rst [hunk 2]
```diff
@@ -92,11 +92,52 @@ Recently, we have made updates to MaskRCNN-ConvNeXt, incorporating the `ConvNeXt
...
+
+.. note::
+
+    To fine-tune better on small labeled datasets used in Semi-SL tasks we adopted a repeat dataset which brought metric improvement in our experiments. However, the training time also increased noticeably.
```
Comment (sovrasov) [inline docs/source/guide/explanation/algorithms/segmentation/instance_segmentation.rst]: To obtain a better performance after fine-tuning, ...
... we adopt ..., which brings ... increases -- for consistency with other sentences

**Review by sovrasov — DISMISSED** [review]

File: src/otx/algorithms/detection/configs/instance_segmentation/efficientnetb2b_maskrcnn/model.py [hunk 3]
```diff
@@ -54,8 +54,8 @@
                 type="DeltaXYWHBBoxCoder", target_means=[0.0, 0.0, 0.0, 0.0], target_stds=[0.1, 0.1, 0.2, 0.2]
             ),
             reg_class_agnostic=False,
-            loss_cls=dict(type="CrossEntropyLoss", use_sigmoid=False, loss_weight=1.0),
-            loss_bbox=dict(type="L1Loss", loss_weight=1.0),
+            loss_cls=dict(type="OrdinaryFocalLoss", gamma=1.5, loss_weight=1.0),
+            loss_bbox=dict(type="SmoothL1Loss", beta=1.0, loss_weight=1.0),
```
Comment (jaegukhyun) [inline src/otx/algorithms/detection/configs/instance_segmentation/efficientnetb2b_maskrcnn/model.py]: This config is for supervised learning, not semisl. Is it intended patch?

Comment (kprokofi) [reply src/otx/algorithms/detection/configs/instance_segmentation/efficientnetb2b_maskrcnn/model.py]: Thank you for this!

**Review by jaegukhyun — APPROVED** [review]

**Review by sovrasov — APPROVED** [review]

**Pull request merged**